src/mo_frontend/typing.ml

```diff
@@ -869,19 +970,21 @@ and check_typ_binds env typ_binds : T.con list * T.bind list * Scope.typ_env * S
   check_typ_bind_sorts env tbs;
   let ts = List.map (fun tb -> tb.T.bound) tbs in
   check_typ_binds_acyclic env typ_binds cs ts;
-  let ks = List.map (fun t -> T.Abs ([], t)) ts in
+  let ks = List.map (fun t -> 
+    let index = generic_variable_index env in
+    T.Abs ([], t, index)) ts in
   List.iter2 (fun c k ->
     match Cons.kind c with
-    | T.Abs (_, T.Pre) -> T.set_kind c k
+    | T.Abs (_, T.Pre, _) -> T.set_kind c k
     | k' -> assert (eq_kind env Source.no_region k k')
   ) cs ks;
   let env' = add_typs env xs cs in
   let _ = List.map (fun typ_bind -> check_typ env' typ_bind.it.bound) typ_binds in
   List.iter2 (fun typ_bind c -> typ_bind.note <- Some c) typ_binds cs;
   cs, tbs, te, T.ConSet.of_list cs
 
-and check_typ_bind env typ_bind : T.con * T.bind * Scope.typ_env * Scope.con_env =
-  match check_typ_binds env [typ_bind] with
+and check_typ_bind env stable_scope typ_bind : T.con * T.bind * Scope.typ_env * Scope.con_env =
```

Comment:
Again, I don't think `stable_scope` enforces anything.

Reply:
Yes, a relict. Thanks for noticing